cli/napi/js_native_api.rs

```diff
@@ -230,9 +234,13 @@ fn napi_define_class<'s>(
     Err(status) => return status,
   };
 
-  let tpl = unsafe {
-    create_function_template(env, Some(name), constructor, callback_data)
-  };
+  let tpl = create_function_template(
+    &mut env.scope(),
+    env_ptr,
+    Some(name),
+    constructor.unwrap(),
```

Comment:
its not because it comes after `check_arg!(env, constructor)`. but yea this gets confusing and wastes performance if rust doesn't optimize it out... i'd like if `check_arg!()` unwrapped it automatically but there's a bit of effort needed there. maybe another followup...